File: backend/FinancialControl.Api/Models/Notification.cs
namespace FinancialControl.Api.Models;

public class Notification
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public Guid? AlertId { get; set; }
    public NotificationType Type { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Message { get; set; } = string.Empty;
    public bool IsRead { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? ReadAt { get; set; }
    public string? ActionUrl { get; set; }
    public string? Metadata { get; set; } // JSON para dados adicionais

    // Navigation properties
    public User User { get; set; } = null!;
    public Alert? Alert { get; set; }
}

public enum NotificationType
{
    Info = 1,
    Warning = 2,
    Error = 3,
    Success = 4,
    Alert = 5
}

File: frontend/src/types/notification.ts
export enum NotificationType {
  Info = 1,
  Warning = 2,
  Error = 3,
  Success = 4,
  Alert = 5,
}

export interface Notification {
  id: string;
  userId: string;
  alertId?: string;
  type: NotificationType;
  title: string;
  message: string;
  isRead: boolean;
  createdAt: string;
  readAt?: string;
  actionUrl?: string;
  metadata?: string;
}

export interface NotificationListItem {
  id: string;
  type: NotificationType;
  title: string;
  message: string;
  isRead: boolean;
  createdAt: string;
  actionUrl?: string;
}

export const notificationTypeColors: Record<NotificationType, string> = {
  [NotificationType.Info]: 'bg-blue-100 text-blue-800',
  [NotificationType.Warning]: 'bg-yellow-100 text-yellow-800',
  [NotificationType.Error]: 'bg-red-100 text-red-800',
  [NotificationType.Success]: 'bg-green-100 text-green-800',
  [NotificationType.Alert]: 'bg-orange-100 text-orange-800',
};

export const notificationTypeIcons: Record<NotificationType, string> = {
  [NotificationType.Info]: '💡',
  [NotificationType.Warning]: '⚠️',
  [NotificationType.Error]: '❌',
  [NotificationType.Success]: '✅',
  [NotificationType.Alert]: '🔔',
};
